src/dialogs/more-info/ha-more-info-dialog.ts

```diff
@@ -695,7 +749,7 @@ export class MoreInfoDialog extends LitElement {
 
         @media all and (min-width: 600px) and (min-height: 501px) {
           ha-dialog {
-            --mdc-dialog-min-width: 580px;
+            --mdc-dialog-min-width: min(100%, 580px);
```

Comment:
580 will always be the min, as the media query only applies when the min width is 600px.